lld/test/wasm/call-indirect-external.s

```diff
@@ -0,0 +1,27 @@
+# RUN: llvm-mc -filetype=obj -triple=wasm32-unknown-unknown -o %t.o %s
```

Comment:
I wonder if we can give this test a better name?   Perhaps `bad-pic-relocations.s`?  

Could we split this test and the corresponding change into a separate PR?

Also, presumably the same error should occur for `R_WASM_MEMORY_ADDR_REL_` relocations against undefined symbols? 

BTW, how were you able to reproduce this error in the wild?   e.g. did it require writing assembly?

Reply:
Thank you. I agree. The error message would also apply to `R_WASM_MEMORY_ADDR_REL_` relocation types.

I have seen that bad pic relocations are reported as errors unless `UnresolvedPolicy::ImportDynamic` is enabled. Perhaps undefined PIC relocations in code should always be reported as errors, since they cannot be later resolved by subsequent dynamic Wasm linking (unlike GOT globals) if I understand correctly. Probably, this is more a design question, and certainly better addressed separately of this PR.

BTW: I noticed the link error when building the runtime system for our PL called Motoko with latest Rust (nigthly 2024-07-28) and `wasm64`.

Reply:
I'm curious how your project was able to generate `MBREL` or `TBREL` relocations types against undefined symbols.  I don't think LLVM should ever do that.. I would expect it would require hand-written assembly.   If you are seeing those in LLVM-generated code then that could point to an LLVM codegen bug.

Reply:
Actually, in my case, the symbols for those relocations were all defined, just not handled by the Wasm linker. The undefined case would only happen if a subset of o-files would be linked (with a permissive `UnresolvedPolicy`).

Reply:
Were the `MBREL`/`TBREL` relocations generated by LLVM?   IIRC those should only be generated for symbols local to a given object file... but perhaps its actually DSO-local which is the deciding factor here.

Are you able to repoduce this issue with C code?  If not then perhaps with llvm bitcode?  Could you share an example that reproduces the issue?

Reply:
I tried to prepare LLVM IR which reproduces it: https://github.com/luc-blaeser/wasm-ld-repro
Originally, I saw the issue when compiling Rust code, and generating a shared Wasm library (https://github.com/luc-blaeser/wasm-ld-repro/blob/main/original_issue.md)
It involves DSO-local.

Reply:
I prepared https://github.com/luc-blaeser/wasm-ld-repro with some description.

Reply:
This is a smaller repro (however this one is hand-written LLVM IR): https://github.com/luc-blaeser/wasm-ld-repro/tree/luc%2Fmini-repro